packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -2063,28 +2079,31 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
     for (_RouteEntry entry in toBeDisposed) {
       entry.dispose();
     }
-    if (SchedulerBinding.instance.schedulerPhase == SchedulerPhase.persistentCallbacks) {
-      SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
+
+    switch (SchedulerBinding.instance.schedulerPhase) {
+      case SchedulerPhase.persistentCallbacks:
+        SchedulerBinding.instance.addPostFrameCallback((Duration duration) {
```

Comment:
Why do we need to do this in a post frame callback?

Reply:
The imperative api might be called during buildstage. For example https://github.com/flutter/flutter/blob/c06bf6503a8b6690b2740a7101852fcc8f133057/packages/flutter/lib/src/material/dropdown.dart#L1249

this works fine prior to the refactor, so I updated this method to be able to work during build stage as well

Reply:
Why did this work without scheduling a post frame callback before?

Reply:
The line you pointed to is illegal. You shouldn't be allowed to mutate the Navigator state during a build, see my comment here: https://github.com/flutter/flutter/pull/42482/files#r359654328

Reply:
Actually, looks like we have special logic in place to make removing an overlay entry during build work: https://github.com/flutter/flutter/pull/42482#discussion_r359656749

Reply:
And this code now allows way more than just removing a route during build. It also allows other mutations (like adding) during build that previously were illegal. I don't think we should extend the functionality like that...

Reply:
Maybe we can get around this by not rearranging the overlay when Routes only get removed. The Route itself is in charge of removing its overlay, so if I am not mistaken this rearrange call should be a no-op.